observability-kit-starter/src/main/java/com/vaadin/observability/ObservabilityHandler.java

```diff
@@ -0,0 +1,105 @@
+package com.vaadin.observability;
+
+import java.io.IOException;
+import java.util.UUID;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import jakarta.servlet.http.HttpServletResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.vaadin.flow.component.ComponentUtil;
+import com.vaadin.flow.component.UI;
+import com.vaadin.flow.server.SynchronizedRequestHandler;
+import com.vaadin.flow.server.VaadinRequest;
+import com.vaadin.flow.server.VaadinResponse;
+import com.vaadin.flow.server.VaadinSession;
+import com.vaadin.flow.shared.ApplicationConstants;
+
+public class ObservabilityHandler extends SynchronizedRequestHandler {
+    private static Logger getLogger() {
+        return LoggerFactory.getLogger(ObservabilityHandler.class);
+    }
+
+    private static final String PATH = "/";
+    private static final String ID_PARAMETER = "id";
+    private static final String REQUEST_TYPE = "o11y";
+    private static final String HTTP_METHOD = "POST";
+    private static final String CONTENT_TYPE = "application/json";
+
+    private final String id = UUID.randomUUID().toString();
+
+    static ObservabilityHandler ensureInstalled(UI ui) {
+        ObservabilityHandler observabilityHandler = ComponentUtil.getData(ui,
+                ObservabilityHandler.class);
+        if (observabilityHandler != null) {
+            // Already installed, return the existing handler
+            return observabilityHandler;
+        }
+
+        ObservabilityHandler newObservabilityHandler =
+                new ObservabilityHandler();
+
+        VaadinSession session = ui.getSession();
+        session.addRequestHandler(newObservabilityHandler);
+        ComponentUtil.setData(ui, ObservabilityHandler.class,
+                newObservabilityHandler);
+
+        ui.addDetachListener(detachEvent ->
```

Comment:
Just a tough, but no suggestion on how to handle currently: in the case the client needs a full resynchronization, detach and attach callbacks are invoked on the UI. So, the request handler may get removed and not added again.
This is actually an issue also with other products, for example MPR, and should perhaps be fixed somehow on the Flow side.
Maybe we can just add a comment to remember that this situation may happen. The workaround is to reload the page.

Reply:
Would the following be enough to ensure the handler is re-installed?

```java
        VaadinSession session = ui.getSession();

        ui.addAttachListener(attachEvent -> {
            session.addRequestHandler(newObservabilityHandler);
            ComponentUtil.setData(ui, ObservabilityHandler.class,
                newObservabilityHandler);
        });
        ui.addDetachListener(detachEvent -> {
            session.removeRequestHandler(newObservabilityHandler);
            ComponentUtil.setData(ui, ObservabilityHandler.class,
                null);
        });
```

Reply:
I think this may be too late on a normal initialization. At least one execution will be skipped.
Perhaps we need to have the registration both in `ensureInstalled` and the listener.
And for the listeners, we need to check if the request handler is already in `session.getRequestHandlers()` before adding it